### PR TITLE
Address the local Administrators group by its well-known SID rather than by name

### PR DESCRIPTION
**Pull request by mkszuba**
Names of security objects built into Windows depend on the installation-time language. For example on a German system the "local administrators" group is called Administratoren rather than Administrators, causing puppet-msoffice to fail because being unable to set the correct owner group on office_config.xml. To make things worse, once set those names cannot be modified even when the default OS language is changed - if you want to rename Administratoren to Administrators you have to reinstall Windows.

Luckily, Puppet now understands Windows security identifiers and there is a well-known SID corresponding to Administrators regardless of the language used. Change _group => 'Administrators'_ to _group => 'S-1-5-32-544'_ and it will work regardless of whether your copy of Windows has been installed in English, German, Swahili, or Klingon.

Nb. Ideally the name of the built-in administrator user should be replaced by a SID too, especially given that in addition to language dependency that name can be changed by a certain Group Policy setting. Unfortunately that one is more tricky because its well-known SID is not fully static and I don't know if Puppet can be told to match user as  _S-1-5-<something>-500_. On the other hand, I have yet to see this particular bit of puppet-msoffice fail even though the built-in administrator account isn't called Administrator on **any** of the machines I work with - so maybe Puppet handles this one internally.

### CODE REVIEW

File: manifests/package.pp [hunk 1]
```diff
@@ -98,7 +98,7 @@
           content => template('msoffice/setup.ini.erb'),
           mode    => '0755',
           owner   => 'Administrator',
```
Comment (Iristyle) [inline manifests/package.pp]: Note that `Administrator` is also localized (and / or renamed).

Comment (mkszuba) [inline manifests/package.pp]: Indeed it is, as I remarked in my opening comment. Thanks for digging up that custom fact! I wonder why I've never seen Puppet complain that one on any of my systems, though.

**Comment by bastelfreak** [comment]
Hi @mkszuba, can you please rebase?

**Comment by mkszuba** [comment]
Done.

**Comment by natemccurdy** [comment]
The fact looks familiar :-)
https://github.com/natemccurdy/windows_account_names/blob/master/lib/facter/windows_accounts.rb

In all seriousness though, I'm not sure it's a good idea to replicate another forge module's fact here.
What is a user has both and I update mine to fix a bug? There will then be a conflict.

I think the better approach is to make a dependency on the module if you do need to use it's fact.

Is it is it required to set the owner to a SID as well?
What if you just parameterized the value for owner and group instead as well?

**Comment by mkszuba** [comment]
> The fact looks familiar :-)

Yup :-) I have simply followed the suggestion made by @Iristyle in the first comment, IIRC the commit message even quotes the exact URI.

> I think the better approach is to make a dependency on the module if you do need to use it's fact.

If you guys are okay with introducing another dependency, definitely.

> Is it is it required to set the owner to a SID as well?
> What if you just parameterized the value for owner and group instead?

As in having the owner and group configurable _e.g._ through Hiera? I must say I am not convinced that would be useful.

On the one hand, can the MS Office installer work with the config file being owned by anyone else than Administrator:Administrators?

On the other, the main reason why I came up with the SID-based solution was that at the time I began using this module, I had to manage Windows instances which had used quite a wide range of installation languages. I reckoned there could be a way of parameterising this even so but it felt like unnecessarily complicating the issue.

**Comment by natemccurdy** [comment]
My vote is to not copy the fact and to instead set a dependency on `nate/windows_account_names`.

Also, adding a module dependency should also mean a bump in the X version of the module. I can understand not wanting to do that.

However, if you are going to use it, might as well use the newer version that has some improvements: https://github.com/natemccurdy/windows_account_names/blob/2.0.0/lib/facter/windows_accounts.rb

And please credit the original module (and me 🍻) in the fact code if you do use it.

**Comment by mkszuba** [comment]
Apologies for an indecently long delay, work caught up with me. I have just updated the second commit so that Nate's module is used to provide the custom fact.

**Comment by bastelfreak** [comment]
@mkszuba can you take a look at the failing CI tests?

**Comment by mkszuba** [comment]
Well, the actual error message has been severely abbreviated so it's hard to see what the exact problem is but as far as as I can see the test suite does not clone the module _windows_account_names_. I have got zero experience with Ruby testing but having poked around the repository for references to modules which do get installed, it seems I should have added _windows_account_names_ to _.fixtures.yml_ in addition to the manifest. Let's see if this helps.

**Pull request closed without merging**